src/v/cloud_storage/partition_recovery_manager.cc

```diff
@@ -292,16 +292,15 @@ partition_downloader::download_log(const remote_manifest_path& manifest_key) {
 
     auto upl_result = co_await _remote->upload_manifest(
       _bucket, target, _rtcnode);
-    // If the manifest upload fails we can't continue
-    // since it will damage the data in S3. The archival subsystem
-    // will pick new partition after the leader will be elected. Then
-    // it won't find the manifest in place and will create a new one.
-    // If the manifest name in S3 matches the old manifest name it will
-    // be overwriten and some data may be lost as a result.
-    vassert(
-      upl_result == upload_result::success,
-      "Can't upload new manifest {} after recovery",
-      target.get_manifest_path());
+
+    // In this case the manifest will be uploaded next time we will roll the
+    // segments
+    if (upl_result != upload_result::success) {
+        vlog(
+          _ctxlog.warn,
+          "Can't upload partition manifest {} after recovery",
+          target.get_manifest_path());
```

Comment:
based on what is happening below it sort of looks like the intention here was to return a `log_recovery_result` rather than also checking if the manifest revision is not the initial revision and potentially changing it. at least returning here would be a big clearer and looks like it would be correct? i guess it all hinges on:

```
    if (mat.topic_manifest.get_revision() != _ntpc.get_initial_revision()) {
        mat.topic_manifest.set_revision(_ntpc.get_initial_revision());
```

Reply:
Are there any issues with failling to upload one manifest, but not the other? I.e. we fail to upload partition manifest but succeed in uploading topic manifest.